packages/data/tests/unit/services/navi-metadata-test.ts

```diff
@@ -0,0 +1,266 @@
+    });
+    await this.service.loadMetadata();
+
+    this.server.get('/tables', function() {
+      assert.notOk(true, 'after metadata is loaded, a fetch request is not executed');
+    });
+  });
+
+  test('all', async function(this: Context, assert) {
+    await this.service.loadMetadata({ dataSourceName: 'bardTwo' });
+    await this.service.loadMetadata({ dataSourceName: 'elideOne' });
+
+    const bardTwoTables = this.service.all('table', 'bardTwo');
+    assert.deepEqual(bardTwoTables.mapBy('id'), ['inventory'], '`all` returns all `elideOne` tables');
+
+    const elideOneTables = this.service.all('table', 'elideOne');
+    assert.deepEqual(elideOneTables.mapBy('id'), ['table0', 'table1'], '`all` return all `elideOne` tables');
+
+    const allTables = this.service.all('table');
+    assert.deepEqual(
+      allTables.mapBy('id'),
+      ['inventory', 'table0', 'table1'],
+      '`all` return all loaded tables when data source is not specified'
+    );
+    assert.ok(
+      allTables.every(table => table instanceof TableMetadataModel),
+      'All returns instances of `TableMetadataModel`s'
+    );
+
+    assert.throws(
+      () => this.service.all('table', 'notLoaded'),
+      /Metadata must have the requested data source loaded: notLoaded/,
+      '`all` throws an error if attempting to get all models from a data source that is not loaded'
+    );
+  });
+
+  test('getById', async function(this: Context, assert) {
+    await this.service.loadMetadata({ dataSourceName: 'elideOne' });
+    await this.service.loadMetadata({ dataSourceName: 'bardTwo' });
+
+    const metricOne = this.service.getById('metric', 'metric1', 'elideOne');
+    assert.ok(
+      metricOne instanceof MetricMetadataModel,
+      '`getById` returns a loaded instance of `MetricMetadataModel` when requesting `metric` type'
+    );
+    assert.equal(metricOne?.name, 'Metric 1', '`getById returns a metadata model given a type, id, & datasource');
+
+    const revenue = this.service.getById('metric', 'revenue', 'bardTwo');
+    assert.ok(
+      revenue instanceof MetricMetadataModel,
+      '`getById` returns a loaded instance of `MetricMetadataModel` when requesting `metric` type'
+    );
+    assert.equal(revenue?.name, 'Revenue', '`getById returns a metadata model given a type, id, & datasource');
+
+    const missing = this.service.getById('metric', 'not a metric', 'bardTwo');
+    assert.equal(missing, undefined, '`getById returns `undefined` if a model by `id` cannot be found');
+
+    const notLoaded = this.service.getById('table', 'table0', 'notLoaded');
+    assert.equal(notLoaded, undefined, '`getById` returns `undefined` when data source is not loaded');
+  });
+
+  test('fetchById - elide', async function(this: Context, assert) {
+    assert.rejects(
+      this.service.fetchById('metric', 'metric1', 'elideOne'),
+      /Type requested in navi-data\/elide-metadata adapter must be defined as a query in the gql\/metadata-queries.js file/,
```

Comment:
```suggestion
      /Type requested in navi-data\/metadata/elide adapter must be defined as a query in the gql\/metadata-queries.js file/,
```
Can we update the file reference